snapshots+code-caching.md

```diff
@@ -0,0 +1,84 @@
+# Snapshots and Code Caching
+
+This document explains techniques used by V8 in order to avoid having to re-compile and
+optimized JavaScript whenever an application that embeds it (i.e. Chrome or Node.js) starts up
+fresh.
+
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+**Table of Contents**  *generated with [DocToc](https://github.com/thlorenz/doctoc)*
+
+- [Code Caching](#code-caching)
+  - [Chrome's Use of Code Caching](#chromes-use-of-code-caching)
+  - [Resources](#resources)
+- [Startup Snapshots](#startup-snapshots)
+  - [Custom Startup Snapshots](#custom-startup-snapshots)
+  - [Lazy Deserialization](#lazy-deserialization)
+  - [Resources](#resources-1)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+## Code Caching
+
+- lessens overhead of parsing + compiling script
+- uses cached data to recreate previous compilation result
+- exposed via V8's API to embedders
+  - pass `v8::ScriptCompiler::kProduceCodeCache` as an option when compiling script
+  - cached data is attached to source object to be retrieved via
+    `v8::ScriptCompiler::Source::GetCachedData`
+  - can be persisted for later
+  - later cache data can be attached to the source object and passed
+    `v8::ScriptCompiler::kConsumeCodeCache` as an option to cause V8 to bypass compileing the
+    code and deserialize the provided cache data instead
+- V8 6.6 caches top level code as well as code generated _after_ script's top-level execution,
```

Comment:
Note that kProduceCodeCache is being deprecated in favor of this new API (ScriptCompiler::GetCodeCache)

Reply:
OK I'll mention that ... also keep in mind that this is meant to be a living document (which is why I made it a github repo instead of a blog).
So please feel free to file issues/PRs when things change in the future so we can get that updated. :)
For now I'll add a note about the coming API change.

Reply:
Addressed in b11e195